### PR TITLE
Added ability for ImageGrid to show RGB images and 4d matrices

### LINKED ISSUE #583: Using map_func on ImageGrid with a list of images modifies the original list
If you provide a list of images to ImageGrid and function in the map_func attribute the function is applied to the original images and they are permanently changed.

Here is some example code to recreate the issue, first we create 2 random images and we visualize them
```python
img1 = np.random.random((32,32))
img2 = np.random.random((32,32))

image_list = [img1, img2]
isns.ImageGrid(image_list)
```
output

Then we use ImageGrid with a function that sets some of the values to 0, notice that the original array is never modified since it is copied before the prodecure

```python
def apply_to_imgs(img):
    replacement = img.copy()
    replacement[5:, :-5] = 0
    return replacement

isns.ImageGrid(image_list, map_func = apply_to_imgs)
```
output2

This is all expected behaviour.

Now if we show the original list again without applying the function, the original images have been modified
```python
isns.ImageGrid(image_list)
```
output3

This is most likely a bug caused by modifying directly the list passed as an input.

### PR DESCRIPTION
**Pull request by eugenioLR**
I changed some code to allow ImageGrid to display a list of RGB images, 4d arrays are now allowed as they can encode n images when they have size (n, x, y, 3). Color images and grayscale images can be also mixed in a list and they will be displayed correctly.

Also, since matplotlib allows displaying images of size (x,y,1) and (x,y,4) it is also allowed here.

I also issue a warning when using a 2d array in ImageGrid since it is prefered that imgplot is used as was sugested in the code.

### CODE REVIEW

**Comment by SarthakJariwala** [comment]
A couple of points regarding documentation: 📜 

- [ ] Please add examples for 4d image and list of 3d images in docstring under `Examples`.  Documentation and example usage will be rendered from there.
- [ ] Update `Muti Dim Images.ipynb` located under `docs/Tutorial/`with new features (4d image, list of 3d images)

Both of these will make for easier feature discovery and usage.

**Comment by SarthakJariwala** [comment]
For purposes of documentation and feature demonstration, if it makes sense to add a new 4d dataset, please feel free to do so. You can take a look at `_datasets.py` file to get an idea about how datasets are currently handled.

**Comment by SarthakJariwala** [comment]
Thanks for this PR @eugenioLR! Excited about this and looking forward to getting this feature merged. 🚀

File: src/seaborn_image/_grid.py [hunk 1]
```diff
@@ -288,7 +288,7 @@ def __init__(
         data,
```
Comment (SarthakJariwala) [inline src/seaborn_image/_grid.py:288]: Update docstring for `data`

File: src/seaborn_image/_grid.py [hunk 2]
```diff
@@ -499,23 +520,40 @@ def _map_img_to_grid(self):
                 _d = self.data[i]
 
                 # check if the image has more than 2 dimensions
-                if _d.ndim > 2:
+                if _d.ndim > 3:
                     raise ValueError(
-                        "can not plot multiple 3D images. Supply an individual 3D image"
+                        "One of the images provided has more than 3 dimensions."
+                        "verify that all the images are correctly formatted."
```
[the remark this answers is not in the document]
Comment (eugenioLR) [reply src/seaborn_image/_grid.py:526]: I agree, that seems like a better error message, i'll add it to the next commit.

**Comment by eugenioLR** [comment]
> For purposes of documentation and feature demonstration, if it makes sense to add a new 4d dataset, please feel free to do so. You can take a look at `_datasets.py` file to get an idea about how datasets are currently handled.

I'm going to add 60 samples from the Cifar10 dataset, the images are a bit blurry but it doesn't look too bad. They will be stored as a numpy array inside a .npy file and it will be 180Kb. I hope that's ok.

**Comment by eugenioLR** [comment]
I think that should address everything, the data is referencing my repository for the moment being since this isn't uploaded yet, when this is pushed to the master branch it should be changed back to this repository.

**Comment by eugenioLR** [comment]
I'm having a bit of trouble with loading the new data file. Can you check it out to see what's wrong?

**Comment by eugenioLR** [comment]
I think that's it. It should be ready to be merged.
The url to the datasets should also be changed to match this repository, but that can be done once everything is uploaded.

**Comment by SarthakJariwala** [comment]
I made some more changes - changed `copy` to be only called when using `map_func`. We don't want to copy if we are not changing the array. Also added a test for it.

**Comment by SarthakJariwala** [comment]
Thanks @eugenioLR for the feature. Merging it and will release as 0.6.0

**Pull request merged**